src/core/reference/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -574,6 +576,111 @@ void InterpolateEval<T>::nearest_func(const T* input_data, T* out) {
     NGRAPH_SUPPRESS_DEPRECATED_END
 }
 
+static void pad_input_data(const uint8_t* data_ptr,
+                           uint8_t* padded_data_ptr,
+                           size_t type_size,
+                           const ov::Shape& input_shape,
+                           const ov::Shape& padded_input_shape,
+                           const std::vector<size_t>& pads_begin) {
+    NGRAPH_SUPPRESS_DEPRECATED_START
+    CoordinateTransform input_transform(input_shape);
```

Comment:
Can we avoid deprecated classes?

Reply:
There hasn't been found yet the alternative class or ways for `CoordinateTransform`. And currently Interpolate-4 is using that class while many other ops using too, such as _avg_pool, batch_norm, logical_reduction, log_softmax, lrn, max, max_pool, mean, min, quantize, roi_align, softmax, sum_ and so on. For now it would better use that class, as it looks conventional and stable.

Reply:
Could you use `CoordinateTransformBasic`?

Reply:
When using `CoordinateTransformBasic`, it produces following error.

> 21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: ========== ERRORS were found in the output:
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: WARNING: --use-feature=2020-resolver no longer has any effect, since it is now the default dependency resolver in pip. This will become an error in pip 21.0.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(593): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(594): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(593): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.
21:53:06  [2022-01-24 15:52:53,219] [5724] build ie ERROR: C:\j\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\core\reference\include\ngraph/runtime/reference/interpolate.hpp(594): error C4996: 'ngraph::CoordinateTransformBasic::index': This method is deprecated and will be removed soon.

Reply:
I think for now it'd better use `CoordinateTransform` and later when it will be deprecated, when all of the parts using `CoordinateTransform` should be modified, Interpolate-1 and 4 also will be modified.